src/main/java/SequenceValidator.java

```diff
@@ -5,167 +5,376 @@
-                        if (!coreqsFound[x]) {
-                            ArrayList<String> affectedCourses = new ArrayList<String>();
-                            affectedCourses.add(courseInfo.prereqs.get(x));
-                            affectedCourses.add(course.getCode());
-                            String message = courseInfo.prereqs.get(x) + " must be taken at least as soon as " + course.getCode();
-                            SequenceIssue issue = new SequenceIssue(message, affectedCourses);
-                            issues.add(issue);
-//                            errors.add(courseInfo.coreqs.get(x) + " must be taken at least as soon as " + course.getCode());
+
+                        // VALIDATE REGULAR ISSUES/WARNINGS
+                        //----------------------------------
+
+                        String courseCode = (String) course.get("code");
+
+                        if (loopCount == 0)
+                        {
+                            if (!courseCode.equals("")) {
+                                if (!cc2ci.containsKey(courseCode)) {
+                                    ArrayList<Point> courseOccurences = new ArrayList<>();
+                                    courseOccurences.add(new Point(uniqueSemesterId, coursesInSemIdx));
+                                    cc2ci.put(courseCode, courseOccurences);
+                                } else {
+                                    cc2ci.get(courseCode).add(new Point(uniqueSemesterId, coursesInSemIdx));
+                                }
+                            }
+
+                            // CREDIT COUNT
+
+                            creditCount += Double.parseDouble((String)course.get("credits"));
+                        }
+                        else if (loopCount == 1)
+                        {
+                            // PREREQUISITE
+
+                            boolean prereqsValid = true;
+                            JSONArray failedRequirements = new JSONArray();
+                            // for each orList of prereqs
+                            JSONArray orLists = null;
+                            try { orLists = course.getJSONObject("requirements").getJSONArray("prereqs");
+                            } catch(Exception e) { continue; /* skip prereq validation if no prerqs */ }
+
+                            for (int i = 0; i < orLists.length(); i++) {
+                                JSONArray orList = orLists.getJSONArray(i);
+
+                                boolean orListValid = false;
+                                // for each course in orList
+                                for (int j = 0; j < orList.length(); j++) {
+                                    String prereqCourseCode = orList.getString(j);
+                                    // prequnisemid = prereq's unique semester id (of first occurence in sequence)
+                                    Integer prequnisemid = null;
```

Comment:
How about `prereqid` instead of `prequnisemid` ?

Reply:
I'll do `prereqsemid`.  I don't like `prereqid` because it's not the ID of the prereq, it's the ID of the semesters that the prereq is in.